src/App.tsx

```diff
@@ -15,35 +16,98 @@ export const goodsFromServer = [
+  ALPHABETICALLY = 'alphabetically',
+  DEFAULT = '',
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType>(SortType.DEFAULT);
+  const [sortOrder, setSortOrder] = useState<boolean>(true);
+
+  function prepareGoods(goods: string[], sortBy: string, order = true) {
+    const preparedGoods = [...goods];
+
+    switch (sortBy) {
+      case SortType.BY_LENGTH: {
+        preparedGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+        break;
+      }
+
+      case SortType.ALPHABETICALLY: {
+        preparedGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+        break;
+      }
+
+      default:
+        break;
+    }
+
+    return order ? preparedGoods : preparedGoods.reverse();
+  }
+
+  function resetHandler(): void {
+    setSortField(SortType.DEFAULT);
+    setSortOrder(true);
+  }
+
+  const visibleGoods = prepareGoods(goodsFromServer, sortField, sortOrder);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== SortType.ALPHABETICALLY,
+          })}
+          onClick={() => setSortField(SortType.ALPHABETICALLY)}
+        >
+          Sort {SortType.ALPHABETICALLY}
         </button>
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
+        <button
+          type="button"
+          className={classNames('button', 'is-success', {
+            'is-light': sortField !== SortType.BY_LENGTH,
+          })}
+          onClick={() => setSortField(SortType.BY_LENGTH)}
+        >
+          Sort by {SortType.BY_LENGTH}
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-warning', {
+            'is-light': sortOrder !== false,
+          })}
+          onClick={() => {
+            setSortOrder(prevOrder => !prevOrder);
+          }}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField !== '' || !sortOrder) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={resetHandler}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          {visibleGoods &&
+            visibleGoods.map(good => (
+              <li key={good} data-cy="Good">
```

Comment:
The `key` prop here should be unique for each item in the list. Using `good` as the key could lead to issues if there are duplicate values in the `goodsFromServer`. Consider using the index of the map function or a unique identifier. This is to comply with [CODE KNOWLEDGE] guidelines.